src/fides/api/alembic/migrations/versions/1e750936a141_migrate_to_fideslang_2_0_uses_and_.py

```diff
@@ -0,0 +1,308 @@
+                    label.replace(key, value) if key in label else label
+                    for key, value in data_label_map.items()
+                    for label in item.get("data_categories", [])
+                ]
+
+        if egress:
+            for item in egress:
+                item["data_categories"] = [
+                    label.replace(key, value) if key in label else label
+                    for key, value in data_label_map.items()
+                    for label in item.get("data_categories", [])
+                ]
+
+        if ingress:
+            update_ingress_query: TextClause = text(
+                "UPDATE ctl_systems SET ingress = :updated_ingress WHERE id= :system_id"
+            )
+            bind.execute(
+                update_ingress_query,
+                {"system_id": row["id"], "updated_ingress": json.dumps(ingress)},
+            )
+
+        if egress:
+            update_egress_query: TextClause = text(
+                "UPDATE ctl_systems SET egress = :updated_egress WHERE id= :system_id"
+            )
+            bind.execute(
+                update_egress_query,
+                {"system_id": row["id"], "updated_egress": json.dumps(egress)},
+            )
+
+
+def update_privacy_declaration_data_categories(
+    bind: Connection, data_label_map: Dict[str, str]
+) -> None:
+    """Upgrade or downgrade data uses from fideslang 2.0 for data categories"""
+    existing_privacy_declarations: ResultProxy = bind.execute(
+        text("SELECT id, data_categories FROM privacydeclaration;")
+    )
+    for row in existing_privacy_declarations:
+        labels = [
+            label.replace(key, value) if key in label else label
+            for key, value in data_label_map.items()
+            for label in row["data_categories"]
+        ]
+
+        update_label_query: TextClause = text(
+            "UPDATE privacydeclaration SET data_categories = :updated_label WHERE id= :declaration_id"
+        )
+        bind.execute(
+            update_label_query,
+            {"declaration_id": row["id"], "updated_label": labels},
+        )
+
+
+def update_ctl_policy_data_categories(
+    bind: Connection, data_label_map: Dict[str, str]
+) -> None:
+    """Upgrade or downgrade data uses from fideslang 2.0 for ctl policies"""
+    existing_ctl_policies: ResultProxy = bind.execute(
+        text("SELECT id, rules FROM ctl_policies;")
+    )
+    for row in existing_ctl_policies:
+        rules: List[Dict] = row["rules"]
+        for i, rule in enumerate(rules or []):
+            data_labels: Dict = rule.get("data_categories", {})
+            for j, val in enumerate(data_labels.get("values", [])):
+                data_category: str = val
+                for key, value in data_label_map.items():
+                    if key in data_category:
+                        data_category = data_category.replace(key, value)
+                rules[i]["data_uses"]["values"][j] = data_category
```

Comment:
ack this replaces matching data use values with data categories!

Reply:
addressed https://github.com/ethyca/fides/pull/3933/commits/9308569ad9e9342d9e43256f65a3f9bd266c20cd